esphome/components/i2c/__init__.py

```diff
@@ -20,6 +22,11 @@
     cv.Optional(CONF_SCAN, default=True): cv.boolean,
 }).extend(cv.COMPONENT_SCHEMA)
 
+I2CMULTIPLEXER_SCHEMA = cv.Schema({
+    cv.Required(CONF_ID): cv.use_id(I2CDevice),
```

Comment:
We want this to specifically be a multiplexer id.

```suggestion
    cv.Required(CONF_ID): cv.use_id(I2CMultiplexer),
```